pkg/gce/gce.go

```diff
@@ -64,18 +64,23 @@ func NewContext() (*Context, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to query gce project-id: %v", err)
 	}
-	ctx.ZoneID, err = ctx.getMeta("instance/zone")
+	myZoneID, err := ctx.getMeta("instance/zone")
 	if err != nil {
 		return nil, fmt.Errorf("failed to query gce zone: %v", err)
 	}
-	if i := strings.LastIndexByte(ctx.ZoneID, '/'); i != -1 {
-		ctx.ZoneID = ctx.ZoneID[i+1:] // the query returns some nonsense prefix
+	if i := strings.LastIndexByte(myZoneID, '/'); i != -1 {
+		myZoneID = myZoneID[i+1:] // the query returns some nonsense prefix
+	}
+	if customZoneID != "" {
+		ctx.ZoneID = customZoneID
+	} else {
```

Comment:
Logic "write default and overwrite IF custom" looks good, thus:
```
if customZoneID != "" {
		ctx.ZoneID = customZoneID
}
```
looks enough. All the "myZoneID" related changes could be removed.